etc/tutorial_status.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+from github import Github, Auth
+
+API_KEY = os.getenv("API_KEY")
+if API_KEY == None:
+    print("API key was not found! Authentication will fail!\nSet the environment variable API_KEY "
+          "to a github access token and try again!\n")
+    exit()
+
+auth = Auth.Token(API_KEY)
+
+g = Github(auth=auth)
+failed=False
+
+try:
+    print("Trying to get jobs.......")
+    # anything that calls out to a network, and might face a transient error
+    repo = g.get_repo("oscar-system/TutorialTesterforOscar")
+    workflow = repo.get_workflow("CI.yml")
+    run = workflow.get_runs()[0]    # get most recent run
+    jobs = run.jobs()
+    print("Done!")
+except Exception as e:
+    print(e)
+    falied = True
+
+resultstring = ""
+
+if not failed:
+    for i in jobs:
+        if i.name == "Prepare Tests":
+            continue
+        print(f"Fetched {i.name}!")
+        name = i.name.split()[-1][0:-1]
+        status = i.conclusion
+        resultstring += f"{name}: {status}\n"
+else:
+    print("Network access failed, resulting file will be unchanged!")
+
+datapath = '/'.join(os.path.abspath(sys.argv[0]).split('/')[0:-2])+'/_data'
+statusfilepath = f"{datapath}/examples_status.yml"
+
+if resultstring=="":
+    # if resultstring is empty, just use the old data
```

Comment:
but we are not using the old state, are we? we are overwriting it now with the content of `example_status.yml` which may be old ?

Moreover, if `resultstring` is *not* empty, we are now discarding it, that seems wrong?

So maybe
```suggestion
if len(resultstring) > 0:
```


But once you do all this, couldn't you remove the `try/except`, the variable `failed` etc. -- and then the script just aborts (doing nothing) if there is an error.

In in `update.sh`, just ignore the exit code

Reply:
It looks plausible.

but I don't underestimate why you then still need to check if resultstring is non-empty?